How-To/Create-a-New-Residential-Feature.md

```diff
@@ -0,0 +1,180 @@
+
+-   Identify which folder the script should go in. Although folders are often self-explanatory, the location may switch throughout the feature creation process. For example, `spatial/spatial-environment-secondary_road.R` is created as a `spatial` feature, but during the transformation step (`proximity.dist_pin_to_secondary_roads.sql)`, it shifts to the `proximity` folder since the metric (distance) is in relation to PINs.
+
+-   Load the required imports for Python or libraries for R.
+
+``` python
+import os
+import boto3
+from dotenv import load_dotenv
+```
+
+``` r
+library(arrow)
+library(aws.s3)
+```
+
+-   Set the `AWS_S3_RAW_BUCKET` environment variable to an S3 bucket where output files will be stored:
+
+```         
+AWS_S3_RAW_BUCKET = os.environ.get("AWS_S3_RAW_BUCKET")
+```
+
+```         
+AWS_S3_RAW_BUCKET <- Sys.getenv("AWS_S3_RAW_BUCKET")
+```
+
+-   Extract the necessary data. This can be done through a few methods: scraping a webpage, utilizing an API, or downloading a file. Be aware that we may need to extract data at different time intervals. For data such as the [Central Business District](https://github.com/ccao-data/data-architecture/blob/master/etl/scripts-ccao-data-raw-us-east-1/spatial/spatial-economy.R), where the geographies are set in stone, the script can reference a geography in a single year. On the other hand, [Secondary Streets](https://github.com/ccao-data/data-architecture/blob/master/etl/scripts-ccao-data-raw-us-east-1/spatial/spatial-environment-secondary_road.R) is structured to download a different set of geographies for every year.
+
+-   Choose the output location based on your transformation strategy. If the file is already clean, and a SQL query can be easily implemented on it, the script should be stored in `ccao-data-warehouse-us-east-1`. If the file needs additional cleaning, place it in `ccao-data-raw-us-east-1`. Use a modification of the following script to upload the data to S3 in python. Note how it joins the aforementioned `AWS_S3_RAW_BUCKET`, as well as identifying the correct prefix, in this case `/housing`, before creating the unique folder `dci`.
+
+```         
+def upload_to_s3(file_content, bucket, key_prefix, file_name):
+    s3 = boto3.resource("s3")
+    object_key = f"{key_prefix}/{file_name}"
+    s3.Bucket(bucket).put_object(Key=object_key, Body=file_content)
+    return f"File uploaded to {bucket}/{object_key}"
+
+upload_to_s3(df, AWS_S3_RAW_BUCKET, "housing/dci", "dci.csv")
+```
```

Comment:
**[Nitpick, non-blocking]** Same as above, this code block could benefit from a `python` language identifier. (I'd add a suggestion for this change, but they don't play well with code blocks, since both suggestions and code blocks use triple-backticks to mark the start and end of blocks.)